contracts/protocol/facets/SequentialCommitHandlerFacet.sol

```diff
@@ -0,0 +1,207 @@
+                ).royaltyInfo(_exchangeId, actualPrice);
+
+                // Verify that fees and royalties are not higher than the price.
+                require((protocolFeeAmount + royaltyAmount) <= actualPrice, FEE_AMOUNT_TOO_HIGH);
+
+                // Get price paid by current buyer
+                uint256 len = sequentialCommits.length;
+                uint256 currentPrice = len == 0 ? offer.price : sequentialCommits[len - 1].price;
+
+                // Calculate the minimal amount to be kept in the escrow
+                escrowAmount = Math.max(actualPrice, protocolFeeAmount + royaltyAmount + currentPrice) - currentPrice;
+
+                // Update sequential commit
+                sequentialCommits.push(
+                    SequentialCommit({
+                        resellerId: buyerId,
+                        price: actualPrice,
+                        protocolFeeAmount: protocolFeeAmount,
+                        royaltyAmount: royaltyAmount
+                    })
+                );
+            }
+
+            // Make sure enough get escrowed
+            if (_priceDiscovery.side == Side.Ask) {
+                if (escrowAmount > 0) {
+                    // Price discovery should send funds to the seller
+                    // Nothing in escrow, need to pull everything from seller
+                    if (tokenAddress == address(0)) {
+                        // If exchange is native currency, seller cannot directly approve protocol to transfer funds
+                        // They need to approve wrapper contract, so protocol can pull funds from wrapper
+                        // But since protocol otherwise normally operates with native currency, needs to unwrap it (i.e. withdraw)
+                        FundsLib.transferFundsToProtocol(address(wNative), seller, escrowAmount);
+                        wNative.withdraw(escrowAmount);
+                    } else {
+                        FundsLib.transferFundsToProtocol(tokenAddress, seller, escrowAmount);
+                    }
+                }
+            } else {
+                // when bid side, we have full proceeds in escrow. Keep minimal in, return the difference
+                if (tokenAddress == address(0)) {
+                    tokenAddress = address(wNative);
+                    if (escrowAmount > 0) wNative.withdraw(escrowAmount);
+                }
+
+                uint256 payout = actualPrice - escrowAmount;
+                if (payout > 0) FundsLib.transferFundsFromProtocol(tokenAddress, payable(seller), payout);
+            }
+        }
+
+        // Since exchange and voucher are passed by reference, they are updated
+        emit BuyerCommitted(exchange.offerId, exchange.buyerId, _exchangeId, exchange, voucher, msgSender());
+        // No need to update exchange detail. Most fields stay as they are, and buyerId was updated at the same time voucher is transferred
+    }
```

Comment:
Return value `receiever` is ignored at this point, since the royalties are distributed at the end during the release of funds. Royalties go to the seller's treasury address set at that point.